### PR TITLE
Drop the PyPI mock dependency

### PR DESCRIPTION
**Pull request by musicinmybrain**
This was only needed for Python 2.7, which `nipy` will no longer support (https://github.com/nipy/nipy/pull/512#discussion_r1333271211).

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -74,10 +74,9 @@ Tests
 =====
 
 To run nipy's tests, you will need to install the nose_ Python testing
-package.  If you are using Python 2.7, you will also need to install the mock_
-testing package - e.g.::
+package::
 
-    pip install nose mock
+    pip install nose
```
Comment (matthew-brett) [inline README.rst:79]:
```suggestion
    pip install nose3
```

File: nipy/info.py [hunk 2]
```diff
@@ -102,10 +102,7 @@
 =====
 
 To run nipy's tests, you will need to install the nose_ Python testing
-package.  If you are using Python 2.7, you will also need to install the mock_
-testing package - e.g.::
-
-    pip install nose mock
+package::
```
Comment (matthew-brett) [inline nipy/info.py:105]:
```suggestion
package::

    pip install nose3

```

**Review by matthew-brett — COMMENTED** [review]
A couple of suggestions.

**Comment by musicinmybrain** [comment]
> A couple of suggestions.

Thanks.

**Pull request merged**